docs/architecture/adr-017-subspace-fee-grant.md

```diff
@@ -0,0 +1,194 @@
+# ADR 017: Subspace fee grant
+
+## Changelog
+
+- Nov 14th, 2022: First draft
+
+## Status
+
+PROPOSED
+
+## Abstract
+
+This ADR introduces a new subspace-based fee grant method, which allows subspace service providers to pay the fees of subspace related transactions for the users.
+
+## Context
+
+Currently, one of the major problems of current Web3 services is complicated to use, it requires users to have tokens paying transaction fees so that they can use the service. For instance, in order to create a post on Desmos-based social network, users have to understand:
+- what `transaction` is and why they need some DSM to broadcast it;
+- get some DSM either via an on-ramp or by swapping existing funds.
+
+The `x/feegrant` gives the possibility to pay fees for the users, meaning that users can use the service without understanding how Web3 works behind. However, the `x/feegrant` allowance is not subspace-specified so subspace fees providers might unexpected pay the fees of the transactions inside another subspace.
+
+## Decision
+
+We will implement a subspace-specified fee grant process base on `x/feegrant` that allows subspace fees providers to pay fees for the users inside the specified subspace. The process of that a subspace fees provider support a user fees will be as follows:
+1. the user asks fees provider a fee grant inside the specified subspace;
+2. the fees provider send a subspace fee grant transaction to Desmos;
+3. Desmos executes the subspace fee grant transaction successfully;
+4. the user can use the service without any tokens inside the specified subspace, the fees will be paid by the fees provider.
+
+```mermaid
+sequenceDiagram
+    participant User
+    participant Fees Provider
+    participant Desmos
+    
+    User->>Fees Provider: 1. ask for a subspace fee grant
+    Fees Provider->>Desmos: 2. send subspace fee grant tx
+    Desmos->>Desmos: 3. execute subspace fee grant tx
+    Desmos-->>Fees Provider:  return ok
+    Fees Provider-->>User: return ok
+
+    User->>Desmos: 4. use app with subspace tx
+    Desmos-->>User: return ok
+```
+
+### DeductFeeDecorator
+
+Currently, `x/auth` provides a `DeductFeeDecorator` based on `x/feegrant` to execute the action deducting fees from the signer/feepayer of a transaction. We will build a new subspace-specified `DeductFeeDecorator` to replace the current one.
+
+The new subspace-specified `DeductFeeDecorator` will operate the fees with the process as follows:
+1. check all the messages in the transaction are the subspace messages and all of them are to the same subspace;
```

Comment:
With this method, could a subspace owner issue a grant for both `MsgSaveProfile` and `MsgCreatePost` messages? If so, could then a user perform a transaction containing one instance of both, without any issue? 

From what I read it seems that these kind of grants are related only to subspace messages, but I think it would be better to allow subspace owners to also issue grants related to `MsgSaveProfile` or other kind of messages since those might be useful when creating a Desmos Profile for the first time within an application.